scrape/manager.go

```diff
@@ -154,10 +156,15 @@ func (m *Manager) reload(t map[string][]*targetgroup.Group) {
 		}
 
 		// Scrape pool doesn't exist so start a new one.
+		m.mtx.RLock()
 		existing, ok := m.scrapePools[tsetName]
+		m.mtx.RUnlock()
+
 		if !ok {
 			sp := newScrapePool(scrapeConfig, m.append, log.With(m.logger, "scrape_pool", tsetName))
+			m.mtx.Lock()
```

Comment:
I find this usage of acquiring the lock multiple times in a loop confusing and wonder if it's really safe in all cases. What about first acquiring one lock, then creating any potentially missing scrape pools, then releasing the lock, and only then calling `Sync()` on each pool?

Reply:
I updated it using your suggestion, but still not completely happy with this piece. 
I think DroppedTargets and Targets should belong in the web handler so I will try to refactor it and if looks better will update the PR.